src/limitedmap.h

```diff
@@ -46,20 +46,22 @@ class limitedmap
         }
         return;
     }
-    void erase(const key_type& k)
+    size_t erase(const key_type& k)
     {
         iterator itTarget = map.find(k);
         if (itTarget == map.end())
-            return;
+            return 0;
         std::pair<rmap_iterator, rmap_iterator> itPair = rmap.equal_range(itTarget->second);
-        for (rmap_iterator it = itPair.first; it != itPair.second; ++it)
+        for (rmap_iterator it = itPair.first; it != itPair.second; ++it) {
```

Comment:
There's an extra unneeded copy of the iterator in the for loop initialization.
I'd rather change to

```
        auto [rmap_it, rmap_end] = rmap.equal_range(itTarget->second);
        for (; rmap_it != rmap_end; ++rmap_it) {
            if(rmap_it->second == itTarget) {
                rmap.erase(rmap_it);
                map.erase(itTarget);
                return 1;
            }
        }

```

Reply:
You're right, I didn't notice, I will fix this... actually, I didn't spend any time refactoring studying/refactoring this class besides the minimal change that I applied, but maybe it's a good chance to also remove the TODO in the code

Reply:
I already examined the TODO and basically I'd remove it. Erase map by iterator removes items in range [begin ... end) (end excluded) hence the comment is misleading (we'd have to advance the iterator).

Reply:
Not resolved but "hopefully" will be done i some future